src/app/services/contracts.service.ts

```diff
@@ -0,0 +1,156 @@
+import {Injectable} from '@angular/core';
+import {HttpClient} from '@angular/common/http';
+import {NGXLogger} from 'ngx-logger';
+import {Observable} from 'rxjs';
+import {environment} from '../../environments/environment';
+import {catchError, filter, map} from 'rxjs/operators';
+import {SnackService} from './snack.service';
+import {StaticValues} from '../static/static-values';
+import {Addresses} from '../static/addresses';
+
+interface Result<T> {
```

Comment:
why you need the interface for single result?
just return T or Array<T>

Reply:
Because result contains a `data` property, and a `status` property. i'll revisit this.

Reply:
Also, the snackservice handleError expects a type, which is not a `T` it's a `{data: T, status: string}` (which is basically the Result type you see)

Reply:
Result type has been altered, as is now no longer ambiguous between `T` and `Array<T>`